pkg/bootstrap/bootstrap.go

```diff
@@ -253,7 +254,9 @@ func (c *BootstrapClient) applyManifests(
 
 	for _, obj := range checkWorkloads {
 		bar.Describe(fmt.Sprintf("Checking Status of %v (%v)", obj.GetName(), obj.GetKind()))
-		if err := c.checkWorkloadReady(obj.GetNamespace(), obj.GetName(), obj.GetKind(), 5*time.Minute); err != nil {
+		ctx, cancel := context.WithTimeout(ctx, 5*time.Minute)
+		defer cancel() // release the context in case checkWorkloadReady returned early
```

Comment:
`defer` inside a loop seems a bit weird to me, also Goland tells me a warning there: "Possible resource leak, 'defer' is called in the 'for' loop". Wouldn't the `cancel` function always be the one of the last loop iteration?

I think this could be made a bit clearer by either introducing a function that's called inside the loop and do the `defer` there, or by calling `cancel()` after `checkWorkloadReady` returns.

Reply:
ah okay, I guess it's not a problem after all: https://go.dev/tour/flowcontrol/13

Reply:
Huh, I actually thought that defer would make the call when the variable scope ends, so TIL. Would this fix the IDE warning?

```go
defer func() { cancel() }()
```

Reply:
Nope, also considered as a possible resource leak

Reply:
should be fixed now